k4GaudiPandora/src/DDSimpleMuonDigi.cc

```diff
@@ -0,0 +1,202 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "DDSimpleMuonDigi.h"
+#include <cctype>
+#include <cstdlib>  // abs
+#include "DD4hep/DD4hepUnits.h"
+#include "DD4hep/Detector.h"
+#include "DDRec/DetectorData.h"
+#include "GaudiKernel/MsgStream.h"
+#include "edm4hep/CalorimeterHit.h"
+#include "edm4hep/Constants.h"
+
+DECLARE_COMPONENT(DDSimpleMuonDigi)
+
+DDSimpleMuonDigi::DDSimpleMuonDigi(const std::string& aName, ISvcLocator* aSvcLoc)
+    : MultiTransformer(aName, aSvcLoc,
+                       {
+                           KeyValues("MUONCollection", {"ECalBarrelCollection"}),
+                           KeyValues("HeaderName", {"EventHeader"}),
+                       },
+                       {KeyValues("MUONOutputCollections", {"CalorimeterHit"}),
+                        KeyValues("RelationOutputCollection", {"RelationMuonHit"})}) {
+  m_uidSvc = service<IUniqueIDGenSvc>("UniqueIDGenSvc", true);
+  if (!m_uidSvc) {
+    error() << "Unable to get UniqueIDGenSvc" << endmsg;
+  }
+
+  m_geoSvc = serviceLocator()->service("GeoSvc");  // important to initialize m_geoSvc
+}
+
+StatusCode DDSimpleMuonDigi::initialize() {
+  //Get the number of Layers in the Endcap and Barrel
+  int layersEndcap = 0, layersBarrel = 0;
+  try {
+    const auto mainDetector = m_geoSvc->getDetector();
+    dd4hep::DetElement                         theDetector = mainDetector->detector(m_detectorNameBarrel);
+    const dd4hep::rec::LayeredCalorimeterData* yokeBarrelParameters =
+        theDetector.extension<dd4hep::rec::LayeredCalorimeterData>();
+
+    layersBarrel = yokeBarrelParameters->layers.size();
+    if (!yokeBarrelParameters) {
+      error() << "oops - yokeBarrelParameters is a null pointer" << endmsg;
+    }
+
+  } catch (std::exception& e) {
+    debug() << "  oops - no Yoke Barrel available: " << e.what() << std::endl;
+  }
+  try {
+    const auto mainDetector = m_geoSvc->getDetector();
+    dd4hep::DetElement                         theDetector = mainDetector->detector(m_detectorNameEndcap);
+    const dd4hep::rec::LayeredCalorimeterData* yokeEndcapParameters =
+        theDetector.extension<dd4hep::rec::LayeredCalorimeterData>();
+    layersEndcap = yokeEndcapParameters->layers.size();
+  } catch (std::exception& e) {
+    debug() << "  oops - no Yoke Endcap available: " << e.what() << std::endl;
+  }
+
+  //If the vectors are empty, we are keeping everything
+  if (m_layersToKeepBarrelVec.size() > 0) {
+    //layers start at 0
+    std::vector<int> layersBarrelVec;
+    layersBarrelVec = std::vector<int>(layersBarrel, 0);
+    if (m_layersToKeepBarrelVec[0] != 0) {
+      for (auto i : layersBarrelVec) {
+        m_useLayersBarrelVec.push_back(false);
+      }
```

Comment:
`m_useLayersBarrelVec` could be initialized using the `std::vector` constructor that you are using above to fill a vector with the same element N times